android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/webextension/GeckoWebExtensionException.kt

```diff
@@ -22,9 +26,37 @@ class GeckoWebExtensionException(throwable: Throwable) : WebExtensionException(t
         internal fun createWebExtensionException(throwable: Throwable): WebExtensionException {
             if (throwable is InstallException) {
                 return when (throwable.code) {
-                    ERROR_USER_CANCELED -> WebExtensionInstallException.UserCancelled(throwable)
-                    ERROR_BLOCKLISTED -> WebExtensionInstallException.Blocklisted(throwable)
-                    else -> GeckoWebExtensionException(throwable)
+                    ERROR_USER_CANCELED -> WebExtensionInstallException.UserCancelled(
+                        extensionName = throwable.extensionName,
+                        throwable,
+                    )
+
+                    ERROR_BLOCKLISTED -> WebExtensionInstallException.Blocklisted(
+                        extensionName = throwable.extensionName,
+                        throwable,
+                    )
+
+                    ERROR_CORRUPT_FILE -> WebExtensionInstallException.CorruptFile(
+                        throwable = throwable,
+                    )
+
+                    ERROR_NETWORK_FAILURE -> WebExtensionInstallException.NetworkFailure(
+                        throwable = throwable,
+                    )
+
+                    ERROR_SIGNEDSTATE_REQUIRED -> WebExtensionInstallException.NotSigned(
+                        throwable = throwable,
+                    )
+
+                    ERROR_INCOMPATIBLE -> WebExtensionInstallException.Incompatible(
+                        extensionName = throwable.extensionName,
+                        throwable,
+                    )
+
+                    else -> WebExtensionInstallException.Unknown(
```

Comment:
I am wondering if we should have the name of the add-on here. I know it _can_ be present but what if it isn't because of a random error that we don't explicitly support above? Wouldn't it be more robust to never consider the name of the extension in all _unknown_ cases?

Reply:
>         val addonName = exception.extensionName ?: "" 


As we are falling back to an [empty string when the add-on name is empty](https://github.com/mozilla-mobile/firefox-android/pull/3332/files/bf56ba952c42b841beb2babd2eddc5993fc0a0c2#diff-c592bd7feeeae61a4cb385f42a7543d328e91f6d2823a05b3f430a8d4c0034f4R122), we are going to end up with the following strings:

1. **When the add-on name is not present** "_Failed to install "_.
2. **When it's present** _"Failed to install add-on name"_.

Reply:
Ok, I guess that's fine then, thanks! (At least in English..)

Reply:
> (At least in English..)

That's a good point, I'll remove the name just to be sure!

Reply:
> Do you think we could add a second string that is only "Failed to install this add-on." when we do not have a extension name?

Yes, let's do that!